src/rez/package_repository.py

```diff
@@ -353,18 +370,26 @@ def _uid(self):
 class PackageRepositoryManager(object):
```

Comment:
Note: These minor changes weren't really needed as part of this PR - this work was done when I started investigating a different direction. In any case, these changes fix a bug, where repositories would've been cached into an LRU pool that's shared across all instances of this class.